src/Columns/ColumnTuple.h

```diff
@@ -118,6 +122,9 @@ class ColumnTuple final : public COWHelper<IColumn, ColumnTuple>
     const ColumnPtr & getColumnPtr(size_t idx) const { return columns[idx]; }
     ColumnPtr & getColumnPtr(size_t idx) { return columns[idx]; }
 
+    /// Empty tuple needs a public method to manage its size.
+    void addSize(size_t delta) { len += delta; }
```

Comment:
Consider using `insertDefault` or `insertManyDefaults` instead

Reply:
But yes. InsertManyDefaults is better.

Reply:
`insertDefault` does the same when `columns` is empty, doesn't it?
https://github.com/amosbird/ClickHouse/blob/c8425996ae8a15c2e97d84bc6b9f1545af2b830d/src/Columns/ColumnTuple.cpp#L195-L200

Reply:
Yes. But it does more when `columns` is not empty. We can use `insertManyDefaults`, which means we'll have to do that with a  columns.empty()` check at the callsite.

Reply:
But I think it's ok, since we've decided to allow inconsistency between `column_length` and `columns[0]->size`